doc/contributing/docs/git.rst

```diff
@@ -0,0 +1,80 @@
+Git workflow
+============
+
+Branching
+---------
+
+Use one branch for a single task, unless you're fixing typos or markup on several pages.
+Long commit histories are hard to manage and sometimes end up stale.
+
+Start a new branch from the last commit on ``latest``.
+Make sure to update your local version of ``latest`` with ``git pull``.
+Otherwise, you may have to rebase later.
+
+Name your branch so it's clear what you're doing. Examples:
+
+*   ``short-issue-description``
+*   ``gh-1234-short-issue-description``
+*   ``your-github-handle/short-issue-description``
+
+Linking issues and PRs
+----------------------
+
+When a PR is linked to an issue:
+
+*   You can go from the issue straight to the PR by clicking the link in the right column.
+*   The issue will be automatically closed when you close the PR.
+
+Specify the issue(s) you want to close in the description of your PR. GitHub will connect them if you use specific
+`keywords <https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword>`__.
+Here are some of them:
+
+*   Closes #1234
+*   Resolves #1234
+*   Fixes #1234
+
+If your PR closes more than one issue, add such a line for each issue.
+
+Commit messages
+---------------
+
+*   In your commit message, convey the nature of the change and possibly the reason why it was made.
```

Comment:
From my experience (and our commit message guidelines supports it) the *reason* is the most important thing. It is what you often can't deduce from the changes.